### PR TITLE
vkd3d: Enable required extensions for OpenXR

### PR DESCRIPTION
**Pull request by yshui**
Same idea as #1964, but for OpenXR this time.

### CODE REVIEW

File: libs/d3d12core/main.c [hunk 1]
```diff
@@ -211,7 +213,42 @@ static char *openvr_device_extensions(DXGI_ADAPTER_DESC *desc)
...
-static uint32_t parse_openvr_extension_list(char *extension_str, char **extension_list)
+static char *openxr_vulkan_extensions(bool device_extensions)
+{
+
+    PFN___wineopenxr_GetVulkanExtensions get_extensions = NULL;
+    const char *sym = device_extensions ? "__wineopenxr_GetVulkanDeviceExtensions"
+                                        : "__wineopenxr_GetVulkanInstanceExtensions";
+    uint32_t len;
+    char *ret = NULL;
+    if (!wineopenxr_module)
```
Comment (HansKristian-Work) [inline libs/d3d12core/main.c:224]: This is not thread safe. Should be done in load_d3d12core as part of the pthread_once work.

File: libs/d3d12core/main.c [hunk 2]
```diff
@@ -211,7 +213,42 @@ static char *openvr_device_extensions(DXGI_ADAPTER_DESC *desc)
     return openvr_device_extensions;
 }
 
-static uint32_t parse_openvr_extension_list(char *extension_str, char **extension_list)
+static char *openxr_vulkan_extensions(bool device_extensions)
+{
+
```
Comment (HansKristian-Work) [inline libs/d3d12core/main.c:218]: Weird newline.

File: libs/d3d12core/main.c [hunk 3]
```diff
@@ -469,8 +506,8 @@ static HRESULT vkd3d_create_instance_global(struct vkd3d_instance **out_instance
         VK_KHR_GET_SURFACE_CAPABILITIES_2_EXTENSION_NAME,
     };
 #ifdef _WIN32
-    uint32_t openvr_extension_count;
-    char *openvr_extensions;
+    uint32_t vr_extension_count;
+    char *openvr_extensions, *openxr_extensions;
```
Comment (HansKristian-Work) [inline libs/d3d12core/main.c:510]: inverse :christmas_tree:

File: libs/d3d12core/main.c [hunk 4]
```diff
@@ -521,9 +562,9 @@ static HRESULT STDMETHODCALLTYPE d3d12core_CreateDevice(d3d12core_interface *cor
 
 #ifdef _WIN32
     struct DXGI_ADAPTER_DESC adapter_desc;
-    uint32_t openvr_extension_count;
+    uint32_t vr_extension_count;
     IDXGIAdapter *dxgi_adapter;
-    char *openvr_extensions;
+    char *openvr_extensions, *openxr_extensions;
```
Comment (HansKristian-Work) [inline libs/d3d12core/main.c:567]: :christmas_tree:

File: libs/d3d12core/main.c [hunk 5]
```diff
@@ -49,6 +49,8 @@ static pthread_once_t library_once = PTHREAD_ONCE_INIT;
 
 #ifdef _WIN32
 static HMODULE vulkan_module = NULL;
+static HMODULE wineopenxr_module = NULL;
+typedef int (WINAPI *PFN___wineopenxr_GetVulkanExtensions)(uint32_t, uint32_t *, char *);
```
Comment (HansKristian-Work) [inline libs/d3d12core/main.c:53]: Why not just PFN_wineopenxr? Double or triple underscore is reserved IIRC, and I don't see a reason to write uglier code than necessary here.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]

File: libs/d3d12core/main.c (hunk 5 above)
Comment (yshui) [reply libs/d3d12core/main.c:53]: Because the original function is called `__wineopenxr_GetVulkanExtensions` in wineopenxr

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**